Source/UrlRequest_Windows.cpp

```diff
@@ -172,37 +168,57 @@ namespace UrlLib
 
         gsl::span<const std::byte> ResponseBuffer() const
         {
-            if (!m_responseBuffer)
+            if (!m_fileResponseBuffer.empty())
             {
-                return {};
+                return {(std::byte*)m_fileResponseBuffer.data(), gsl::narrow_cast<std::size_t>(m_fileResponseBuffer.size())};
```

Comment:
Is `gsl::narrow_cast` necessary? They are the same type?
```suggestion
                return {(std::byte*)m_fileResponseBuffer.data(), m_fileResponseBuffer.size()};
```